### ISSUE TITLE
[v1] SSR - add option to extract CSS for this mode too

### ISSUE THREAD

**Issue by kenberkeley** [quasarframework/quasar#3883]
### Software version

Quasar: 1.0.0-beta.13
@quasar/app (v1+ only): 1.0.0-beta.14
OS: OS X 10.11.6
Node: 10.4.1
NPM: 6.4.1
Browsers: Chrome 73.0
iOS: null
Android: null
Any other software related to your bug: null

### What did you get as the error?
According to https://v1.quasar-framework.org/quasar-cli/quasar-conf-js#build-Property
![image](https://user-images.githubusercontent.com/13287948/56079408-5263df80-5e37-11e9-83fb-a6b2399eb293.png)

Here is my config:
```js
// quasar.config.js
{
  ...
  build: {
    distDir: '.dist',
    vueRouterMode: 'history',
    gzip: false, // use Nginx only
    scopeHoisting: true,
    vueCompiler: false,
    analyze: ctx.prod,

    extractCSS: ctx.prod, // <-- HERE!
    sourceMap: false,

    extendWebpack (cfg) {...}
  },
  ...
}
```

After running `quasar build -m ssr` and run the dist:
![image](https://user-images.githubusercontent.com/13287948/56079450-e3d35180-5e37-11e9-9725-ced5b80498ba.png)
![image](https://user-images.githubusercontent.com/13287948/56079473-57755e80-5e38-11e9-8670-a2338f467eaf.png)
![image](https://user-images.githubusercontent.com/13287948/56079486-8db2de00-5e38-11e9-9c99-da488367270d.png)

### What were you expecting?
All the styles should be extracted into `vendor.[hash].css` and `app.[hash].css`, instead of bloating the template.

**Comment by kenberkeley** [comment]
Or, I have to do these myself according to https://ssr.vuejs.org/guide/css.html#css-management ?

**Comment by kenberkeley** [comment]
I thought this feature is out-of-box.

**Comment by rstoenescu** [comment]
When dealing with SSR mode, the server bundle MUST inline critical path CSS so Quasar overrides the setting to ensure it gets the correct value. This is a feature, it's by design.

**Comment by kenberkeley** [comment]
Wait... I also found tons of styles in `vendor.[hash].js` and `app.[hash].js`

![image](https://user-images.githubusercontent.com/13287948/56079615-665d1080-5e3a-11e9-8ee4-0102513ed558.png)

![image](https://user-images.githubusercontent.com/13287948/56079620-7aa10d80-5e3a-11e9-87b6-99beb8bdfbc1.png)

**Comment by rstoenescu** [comment]
Yes, this is by design. Any other way and your ssr app would break.

**Comment by kenberkeley** [comment]
I built a SSR website with Nuxt.js before, and Nuxt.js can extract CSS files as normal...
I am not sure why Quasar can not...

**Comment by kenberkeley** [comment]
I mean, it's not necessary to inline CSS in the template...

**Comment by kenberkeley** [comment]
One more thing, I found some duplicate styles:

![image](https://user-images.githubusercontent.com/13287948/56079786-53e3d680-5e3c-11e9-8069-a97776327fb3.png)
![image](https://user-images.githubusercontent.com/13287948/56079791-69f19700-5e3c-11e9-8bc9-1fe3f57f2bc7.png)

**Comment by rstoenescu** [comment]
Will do some research again with the latest version of Vue and if everything checks out I'll add an option to extractCSS for SSR too. Will post updates this weekend.

**Comment by kenberkeley** [comment]
We can't optimize (dedupe & compress & purge unused) styles without extraction.

**Comment by kenberkeley** [comment]
For example, I built [61 Financial News](https://news.61financial.com.au)'s UI with [Bulma](https://bulma.io):
![image](https://user-images.githubusercontent.com/13287948/56079897-c608eb00-5e3d-11e9-8d9a-601d044642a0.png)

With the help of [PurgeCSS](https://www.purgecss.com), the final build is:
![image](https://user-images.githubusercontent.com/13287948/56079908-fa7ca700-5e3d-11e9-9733-458176d91769.png)
Which is great!

By the way, I appreciate your awesome work of Quasar, I am building a large App with it.

**Comment by matamune94** [comment]
i can't extract CSS too, need fix it.

**Comment by ivanjuras** [comment]
Can we get the option to extract CSS for SSR, please?

**Comment by rstoenescu** [comment]
Will be available in quasar/app v1.1.1 by default.

**Comment by rstoenescu** [comment]
"quasar/app" v1.1.1 released.
